particles/Processing/js/ResizeBilinear.js

```diff
@@ -0,0 +1,25 @@
+/**
+ * @license
+ * Copyright (c) 2019 Google Inc. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * Code distributed by Google as part of this project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+defineParticle(({DomParticle, resolver, log}) => {
+
+  importScripts(resolver(`$here/tf.js`));
+
+  return class extends self.TfMixin(DomParticle) {
+    async update({images, size, options}) {
+      if (images && images.ref && size) {
```

Comment:
hmm... should we be checking that `ref` is defined elsewhere and not just here? 
What if `images.ref` is zero?

Reply:
Good catch, I didn't see this through. Checking .ref everywhere is problematic, I'd rather have an invariant that ref is never falsey. I'll make sure the reference generator is 1-based.